spring-cloud-gcp-pubsub/src/main/java/org/springframework/cloud/gcp/pubsub/core/PubSubTemplate.java

```diff
@@ -124,4 +160,11 @@ public void setMessageConverter(MessageConverter messageConverter) {
 		this.messageConverter = messageConverter;
 	}
 
+	public int getExecutorThreadCount() {
+		return this.executorThreadCount;
+	}
+
+	public void setExecutorThreadCount(int executorThreadCount) {
+		this.executorThreadCount = executorThreadCount;
```

Comment:
@meltsufin ,

what is the reason if that can be done externally and an appropriate `ExecutorProvider` will be injected here?

I don't mind about something default internally, but programming interface of this class (contract) should be as transparent as possible. And that way we sever responsibility and chase the flexibility of the dependency injection.